### ISSUE TITLE
reproducing the results

### ISSUE THREAD

**Issue by yinjiexidian** [HHU-MMBS/TEMI-official-BMVC2023#1]
Thank you for your excellent work. I have a small issue reproducing the article's results, as my performance is lower than expected (cifar10: 87.60, cifar100: 64.22). I ran python gen_embeds.py --arch dino_vitb16 --dataset CIFAR10 --batch_size 256, 

and then, export CUDA_VISIBLE_DEVICES=0; outdir=$"./experiments/TEMI-output-test" ; clusters=10 ; dataset=$"CIFAR10"; 
python train_main.py  --precomputed --arch dino_vitb16  --batch_size=1024 --use_fp16=false --max_momentum_teacher=0.996 --lr=1e-4 --warmup_epochs=20 --min_lr=1e-4 --epochs=100 --output_dir $outdir --dataset $dataset  --knn=50 --out_dim=$clusters  --num_heads=16 --loss TEMI --loss-args beta=0.6. I would like to know if I missed any steps or parameters.

**Comment by black0017** [comment]
Hi @yinjiexidian,

The readme contains an example that runs the code. I didn't enter the sota results but it's usually quite insensitive to the choice of hyperparameters.

For sota results, compare the hyperparameters with the attached json file that we used to report the paper results, mostly batch size, lr, epochs etc....

[hp.json.zip](https://github.com/HHU-MMBS/TEMI-official-BMVC2023/files/12845703/hp.json.zip)

Let me know if you can get the expected accuracies on CIFAR100 (around 67% acc) and CIFAR10. On IN1K we ve used slightly different parameters as mentioned in the paper. 

Best regards and thank you for the nice words!

Nikolas

**Comment by yinjiexidian** [comment]
Thank you very much for your reply. After modifying the parameters according to the JSON file you provided, I achieved the SOTA results in the paper. Thank you again for your excellent work.

**Comment by black0017** [comment]
Thank you so much! I will add a folder with the sota hyperparameters for imagenet and cifar.

**Comment by yinjiexidian** [comment]
Sorry to bother you again. I found that when I did not use the --precomputed parameter, that is, when I passed the image through the network to extract features every time, I got suboptimal results. For example, the clustering accuracy of cifar100 was about 62~63. When I debugged, I found that different functions were used to calculate emb, such as “def backbone_embed(self, x) ”and “def compute_neighbors(embedding, k)”. I wonder where I went wrong?
